# HG changeset patch
# User Rik <rik@octave.org>
# Date 1430352564 25200
#      Wed Apr 29 17:09:24 2015 -0700
# Node ID 3797df9219887163675da937e707536d0cfe55ce
# Parent  4ff975c58584e4bd158fb23845affa2087d05c4d
close.m: Silently accept null argument for Matlab compatibility.

* close.m: Check isempty (arg1) and don't complain if arg1 is null.

diff --git a/scripts/plot/util/close.m b/scripts/plot/util/close.m
--- a/scripts/plot/util/close.m
+++ b/scripts/plot/util/close.m
@@ -66,16 +66,18 @@ function retval = close (arg1, arg2)
       figs = [];
     endif
   elseif (nargin == 1)
     if (ischar (arg1) && strcmpi (arg1, "all"))
       figs = (get (0, "children"))';
       figs = figs(isfigure (figs));
     elseif (any (isfigure (arg1)))
       figs = arg1(isfigure (arg1));
+    elseif (isempty (arg1))
+      figs = [];  # Silently accept null argument for Matlab compatibility
     else
       error ('close: first argument must be "all" or a figure handle');
     endif
   elseif (ischar (arg2)
           && (strcmpi (arg2, "hidden") || strcmpi (arg2, "force")))
     if (ischar (arg1) && strcmpi (arg1, "all"))
       figs = (allchild (0))';
       figs = figs(isfigure (figs));
